tide/master/rest/RestWindows.cpp

```diff
@@ -0,0 +1,143 @@
+
+#include "ContentLoader.h"
+#include "control/ContentWindowController.h"
+#include "thumbnail/thumbnail.h"
+
+#include <tide/master/version.h>
+
+#include <QBuffer>
+#include <QDateTime>
+#include <QHostInfo>
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+
+namespace
+{
+QRegExp _regex = QRegExp("\\{|\\}");
+}
+
+RestWindows::RestWindows( RestServer& server, const MasterConfiguration& config,
+                          const DisplayGroup& displayGroup)
+    : httpServer ( server ), _config( config ),_displayGroup( displayGroup )
+
+{
+    connect( &displayGroup, &DisplayGroup::contentWindowAdded,
+              this, &RestWindows::registerThumbnailUrl );
+    connect( &displayGroup, &DisplayGroup::contentWindowRemoved,
+              this, &RestWindows::deregisterThumbnailUrl );
+}
+
+std::string RestWindows::getTypeName() const
+{
+    return _name;
+}
+
+std::string RestWindows::_toJSON() const
+{
+    std::string content;
+
+    QJsonObject obj;
+    const auto& windows = _displayGroup.getContentWindows();
+
+    QJsonArray arr ;
+
+    for (const auto& element : windows)
+    {
+
+        if (element.get()->getContent().get()->getURI() == "Launcher")
+            continue;
+
+       QJsonObject window ;
+       ContentWindowController controller( *( element.get() ), _displayGroup );
+
+       window["aspectRatio"] = element.get()->getContent()->getAspectRatio();
+       window["minWidth"] = controller.getMinSizeAspectRatioCorrect().width();
+       window["minHeight"] = controller.getMinSizeAspectRatioCorrect().height();
+       window["width"]=( element.get()->getDisplayCoordinates().width( ));
+       window["height"]=( element.get()->getDisplayCoordinates().height( ));
+       window["x"]=( element.get()->getDisplayCoordinates().x( ));
+       window["y"]=( element.get()->getDisplayCoordinates().y( ));
+       window["z"]=(_displayGroup.getZindex( element ));
+       window["title"]=( element.get()->getContent().get()->getTitle( ));
+       window["mode"]=( element.get()->getMode( ));
+       window["selected"]=( element.get()->isSelected( ));
+       window["fullscreen"]=( element.get()->isFullscreen( ));
+       window["focus"]=( element.get()->isFocused( ));
+       window["uri"]=( element.get()->getContent().get()->getURI( ));
+       window["uuid"]= element.get()->getID().toString().replace( _regex,"" );
+       arr.append(window);
+    }
+    obj["windows"] = arr;
+
+    QJsonDocument doc( obj );
+    QString strJson( doc.toJson( QJsonDocument::Indented ));
+    content.append( strJson.toStdString( ));
+
+    return content;
+}
+
+void RestWindows::registerThumbnailUrl( ContentWindowPtr contentWindow )
+{
+    QString url = "tide/windows/" + contentWindow.get()->getID().toString()
+            .replace(_regex,"")+"/thumbnail";
+    QString uri = contentWindow.get()->getContent().get()->getURI();
+    httpServer.get().handleGET(url.toStdString(), [uri] ()
+    {
+        QSize size(768,768);
```

Comment:
magic number: it must be declared as a constant in the namespace{} at the beginnig of the file